File: scripts/python/increment.py
import configparser
import time

config = configparser.ConfigParser()
config.read('../etc/main.conf')
debug = int(config['global']['DEBUG'])

if __name__ == "__main__":
    if debug:
        print("Python (increment) in DEBUG mode...")
    else:
        print("Python (increment)... please wait")

    t0, l_inc, i = time.time(), 0, 0
    step, final = int(config['increment']['STEP']), int(config['increment']['FINAL'])

    while i < final:
        debug and i % step == 0 and print("\rincrementing... %.1f%%" %( i*100 / final ), end='', flush=True )
        l_inc += len(str(i))
        i += 1
    debug and print()

    print("increment %d times, total length %d" %( final, l_inc ) )
    print("time total: %.3f seconds" %( time.time() - t0 ) )

File: scripts/perl/increment.pl
use strict;
use warnings;

use Time::HiRes 'time';
use Config::Tiny;

$| = 1;
my $config = Config::Tiny->read( '../etc/main.conf' );
my $DEBUG = $config->{global}->{DEBUG};

printf "Perl (increment)%s...%s\n",
       $DEBUG ? ' in DEBUG mode' : '',
       $DEBUG ? '' : ' please wait';

my ($i, $l_inc) = (0, 0);
my ($step, $final) = ($config->{increment}->{STEP}, $config->{increment}->{FINAL});
my $t0 = time;

while ($i < $final) {
    $DEBUG and ($i % $step == 0) and printf("\rincrementing... %.1f%%", $i*100/$final);
    $l_inc += length($i);
    $i++;
}
print "\n"
    if $DEBUG;

printf "increment %d times, total length %d\n",
    $final, $l_inc;

printf "time total: %.3f seconds\n", time - $t0;
